src/main/java/org/snomed/snowstorm/core/data/domain/QueryConcept.java

```diff
@@ -299,10 +300,22 @@ private static String serializeMap(Map<Integer, Map<String, List<Object>>> group
 					builder.append(type);
 					builder.append("=");
 					for (Object value : attributes.get(type)) {
-						if (!(value instanceof String)) {
+						String valueString;
+						if (value instanceof String) {
+							valueString = (String) value;
+						} else {
 							builder.append("#");
+							if (value instanceof Double || value instanceof Float) {
+								// Maximum decimal places is 6 - See https://confluence.ihtsdotools.org/display/mag/Concrete+Domain+Decimal+Places+and+Rounding
+								valueString = new DecimalFormat("#0.0#####").format(value);
+							} else {
+								valueString = value.toString();
+							}
```

Comment:
This fix doesn't work for me @kaicode. I created a concrete concept on a local instance, but I'm still getting "5.0E-4 is not a valid number". Is this fix for future concept creations or should it work for historical data (my test concept) too?

Reply:
The fix requires a rebuild of the semantic index. It will also work on new concepts without rebuilding.
`POST /admin/{branch}/actions/rebuild-semantic-index`
The semantic index rebuild should at the base of the branch tree (MAIN) and the change rebased into child branches.

For extensions the International Release branch should have a semantic index rebuild, then the extension working branches should be rebased. Snowstorm will block rebase of extension branches because the upgrade function should be used. However I noticed in the code that the upgrade function has been changed to block upgrading to the same international release again - effectively blocking data fixes traveling into extension branches. 

In the short term the block could be removed in a local Snowstorm version and the upgrade performed in each environment by pointing that local Snowstorm instance at each Elasticsearch environment.
In the medium term I would suggest removing this block from the Snowstorm API level and implementing it in UI logic instead.

Reply:
Thinking again - if this issue only exists in one extension then the semantic index rebuild only needs to happen there. It could be run on that working branch and rebased into child branches.